contracts/global-extensions/GlobalTradeExtension.sol

```diff
@@ -0,0 +1,166 @@
+     *
+     * @param _delegatedManager     Instance of the DelegatedManager to initialize
+     */
+    function initializeExtension(IDelegatedManager _delegatedManager) external onlyOwnerAndValidManager(_delegatedManager) {
+        require(_delegatedManager.isPendingExtension(address(this)), "Extension must be pending");
+
+        ISetToken setToken = _delegatedManager.setToken();
+
+        _initializeExtension(setToken, _delegatedManager);
+
+        emit TradeExtensionInitialized(address(setToken), address(_delegatedManager));
+    }
+
+    /**
+     * ONLY OWNER: Initializes TradeExtension to the DelegatedManager and TradeModule to the SetToken
+     *
+     * @param _delegatedManager     Instance of the DelegatedManager to initialize
+     */
+    function initializeModuleAndExtension(IDelegatedManager _delegatedManager) external onlyOwnerAndValidManager(_delegatedManager){
+        require(_delegatedManager.isPendingExtension(address(this)), "Extension must be pending");
+
+        ISetToken setToken = _delegatedManager.setToken();
+
+        _initializeExtension(setToken, _delegatedManager);
+        _initializeModule(setToken, _delegatedManager);
+
+        emit TradeExtensionInitialized(address(setToken), address(_delegatedManager));
+    }
+
+    /**
+     * ONLY MANAGER: Remove an existing SetToken and DelegatedManager tracked by the TradeExtension
+     */
+    function removeExtension() external override {
+        IDelegatedManager delegatedManager = IDelegatedManager(msg.sender);
+        ISetToken setToken = delegatedManager.setToken();
+
+        _removeExtension(setToken, delegatedManager);
+    }
+
+    /**
+     * ONLY OPERATOR: Executes a trade on a supported DEX.
+     * @dev Although the SetToken units are passed in for the send and receive quantities, the total quantity
+     * sent and received is the quantity of SetToken units multiplied by the SetToken totalSupply.
+     *
+     * @param _setToken             Instance of the SetToken to trade
+     * @param _exchangeName         Human readable name of the exchange in the integrations registry
+     * @param _sendToken            Address of the token to be sent to the exchange
+     * @param _sendQuantity         Units of token in SetToken sent to the exchange
+     * @param _receiveToken         Address of the token that will be received from the exchange
+     * @param _minReceiveQuantity   Min units of token in SetToken to be received from the exchange
+     * @param _data                 Arbitrary bytes to be used to construct trade call data
+     */
+    function trade(
+        ISetToken _setToken,
+        string memory _exchangeName,
+        address _sendToken,
+        uint256 _sendQuantity,
+        address _receiveToken,
+        uint256 _minReceiveQuantity,
+        bytes memory _data
+    )
+        external
+        onlyOperator(_setToken)
+        onlyAllowedAsset(_setToken, _receiveToken)
+    {
```

Comment:
can add `mutualUpgrade(methodologist, operator)` here to get multi-sig `trade()` functionality